feat(#2663): support for playwright-core usage

andremoah opened the pull request:
## Motivation/Description of the PR
- Resolves #2663 .

Added a simple utility that allows the require fallback of npm packages. If the first is not found it will try the next one in the chain.
I have been using it to include playwright-core instead of playwright because I don't want to download the browser binaries on yarn/npm install. This approach can be used for other helpers as well if you guys think it's worth it.

For example, it would be possible as well for puppeteer without adding 

```
npm config set puppeteer_skip_chromium_download true
```

as explained in https://github.com/codeceptjs/CodeceptJS/issues/2076
 

Applicable helpers:

- [ ] WebDriver
- [ ] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] FileHelper
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe
- [x] Playwright

Applicable plugins:

- [ ] allure
- [ ] autoDelay
- [ ] autoLogin
- [ ] customLocator
- [ ] pauseOnFail
- [ ] coverage
- [ ] retryFailedStep
- [ ] screenshotOnFail
- [ ] selenoid
- [ ] stepByStepReport
- [ ] stepTimeout
- [ ] wdio
- [ ] subtitles

## Type of change

- [ ] :fire: Breaking changes
- [x] :rocket: New functionality
- [ ] :bug: Bug fix
- [ ] :clipboard: Documentation changes/updates
- [ ] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [x] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

DavertMik commented:
Thanks, this is a good idea. Probably the installation of playwright-core should be documented as well...
Does this solution is also applicable for puppeteer & puppeteer code?

andremoah commented:
> Thanks, this is a good idea. Probably the installation of playwright-core should be documented as well... Does this solution is also applicable for puppeteer & puppeteer code?

Yes totally! the logic is exactly the same, I can do some extended tests to see if it behaves the same way as Playwright. About the documentation, do you mean in https://codecept.io/ or in the code? I am open to contributing to that, just please let me know  how can i help best.

DavertMik commented:
I think you can update documentation inside the code of Playwright helper. In the very beginning of a file.
And yes, please make the same fix (with the same docs update) for Puppeteer

andremoah commented:
> I think you can update documentation inside the code of Playwright helper. In the very beginning of a file. And yes, please make the same fix (with the same docs update) for Puppeteer

Updated docs and added also puppeteer-core support.

DavertMik commented:
Thank you!

DavertMik commented on lib/helper/Playwright.js:
````diff
@@ -49,12 +50,16 @@ const { createValueEngine, createDisabledEngine } = require('./extras/Playwright
  *
  * ```
  * npm i playwright@^1 --save
+ *
````
```suggestion
 * ```
 * or
 * ```
```

DavertMik commented on lib/helper/Playwright.js:
````diff
@@ -49,12 +50,16 @@ const { createValueEngine, createDisabledEngine } = require('./extras/Playwright
+ * npm i playwright-core@^1 --save
  * ```
  *
+ * Using the core version of playwright package, will prevent download of browser binaries and allows you to connect to an existing browser installation or for connecting to a remote one.
````
```suggestion
 * Using playwright-core package, will prevent the download of browser binaries and allow connecting to an existing browser installation or for connecting to a remote one.
```

DavertMik commented on lib/helper/Puppeteer.js:
````diff
@@ -43,7 +44,13 @@ const consoleLogStore = new Console();
+ * Requires `puppeteer` or `puppeteer-core` package to be installed.
+ * ```
+ * npm i puppeteer --save
+ *
````
```suggestion
 * ```
 * or
 * ```
```

DavertMik commented on lib/helper/Puppeteer.js:
````diff
@@ -43,7 +44,13 @@ const consoleLogStore = new Console();
+ *
+ * npm i puppeteer-core --save
+ * ```
+ * Using the core version of puppeteer package, will prevent download of browser binaries and allows you to connect to an existing browser installation or for connecting to a remote one.
````
```suggestion
 * Using `puppeteer-core` package, will prevent the download of browser binaries and allow connecting to an existing browser installation or for connecting to a remote one.
```
